aiida/manage/configuration/profile.py

```diff
@@ -127,33 +128,43 @@ def __init__(self, name, attributes, from_config=False):
         # Currently, whether a profile is a test profile is solely determined by its name starting with 'test_'
         self._test_profile = bool(self.name.startswith('test_'))
 
-    def get_repository_container(self) -> 'Container':
-        """Return the container of the profile's file repository.
+    @property
+    def _repository_container_path(self) -> str:
```

Comment:
paths should be return as `pathlib.Path`
if not already the case, it would be good if `Container` accepted a `Path` for its initialisation

Reply:
One of my comments as well 😅

Reply:
In my (very poor) defence, I only copy-pasted @giovannipizzi's code snippets from slack and tested if it worked. 😅

Reply:
Those are not meant as the final version to go into the codebase I'm suspecting ;) So that's a "no" on your defence 😂

Reply:
Also, @mbercx if you haven't already, you should install `pre-commit` in the repo by running `pre-commit install`. If it says you don't have `pre-commit` install with the `pre-commit` extra, i.e., `pip install -U -e .[pre-commit]` (or similar).

Reply:
I have the pre-commit hooks installed, and they all passed for my commit. 🤷‍♂️  I haven't run `pre-commit install` in a while though, maybe the hooks have changed since then? I didn't know there should be an empty line after an import in a method (I'm assuming you're referring to [this comment](https://github.com/aiidateam/aiida-core/pull/4889#discussion_r622917362)).

Reply:
Indeed. I thought we were using black now. i guess not? But black always puts an empty new line after imports.

Reply:
the `Container` calls `os.path.realpath` on the path argument. Does that support `Path` objects already in Python 3.6? If so, then we are good.

Reply:
> Edit: Ah, no. We're indeed not yet using black. Still on yapf.... pff

Ah you are preaching to the choir my friend 😆: https://github.com/aiidateam/aiida-core/pull/4345#issuecomment-826762603

> you should install pre-commit in the repo by running pre-commit install

you could really do that with aiida-core, because pylint and mypy use system installs: use tox (its the way forward): `tox -e py38-pre-commit`

Reply:
>  Does that support Path objects already in Python 3.6?

I think everything in `os.path` supports `Path` for >=3.6, but can check

Reply:
yep:

```python
In [5]: realpath??
Signature: realpath(filename)
Source:   
def realpath(filename):
    """Return the canonical path of the specified filename, eliminating any
symbolic links encountered in the path."""
    filename = os.fspath(filename)
    path, ok = _joinrealpath(filename[:0], filename, {})
    return abspath(path)
File:      ~/opt/miniconda3/envs/python-3.6/lib/python3.6/posixpath.py
Type:      function

In [6]: realpath(Path("."))
Out[6]: '/Users/chrisjsewell/Documents/GitHub/aiida-core-origin'
```